src/processes/antiraidListener.ts

```diff
@@ -16,29 +16,45 @@ export const antiraidListener: ProcessInterface = {
         const mentions = new Map()
         const mutedRole = config.roleIDs.muted
 
-        // TODO: fix staffRole to mention properly, cannot be string
-        //const staffRole = config.roleIDs.staff
-        const staffRole = 662405675107876864
+        // NOTE: Pull from config, same role is achieved
+        const staffRole = config.roleIDs.staff
+        // const staffRole = 662405675107876864
+
+        // NOTE: Get role from config
+        const seniorRole = config.roleIDs.senior
+
+        // DEBUG: Debug console log for role comparison
+        // console.log(staffRole, seniorRole)
 
         client.on('message', async (message: Message) => {
             const lastKnownTimestamp = mentions.get(message.author.id) ?? null
 
-            // TODO: escape check if user is senior
-            //message.member.roles
-
-            const {
-                users: { size: mentionedUsers },
-                roles: { size: mentionedRoles }
-            } = message.mentions
-
-            if (
-                mentionedRoles > 1 ||
-                mentionedUsers > 2 ||
-                mentionedUsers + mentionedRoles > 2
-            ) {
-                lastKnownTimestamp
-                    ? checkOffender(message, lastKnownTimestamp)
-                    : addOffenderToList(message)
+            const userRoles = message.member.roles.cache
+
+            // Staff role is not used as of now
+            const staff =
+                userRoles.find(role => role.id === staffRole) != undefined
+            const senior =
+                userRoles.find(role => role.id === seniorRole) != undefined
+
+            // NOTE: Cant combine the two roles else theres weird operator logic
+            if (!senior) {
+                if (!staff) {
```

Comment:
```suggestion
if (!senior || !staff) {
```
?

Reply:
Also
```js
if (isSenior || isStaff) {
  return
}
```

Reply:
Doesn't work due to if there is a staff member that isn't a senior then it prunes the message.
So say that ``Frank`` has the roles ``Staff`` but not ``Senior`` then the if statement evaluates to ``true || false`` resulting in ``true`` going through the statement. This is why it needs to be seperated out into two different if statements.

Reply:
@austinmccalley Check what @waffln said? Thats the solution.

Reply:
@austinmccalley If the user is senior, or he is staff, return (therefore aborting raid-check).